aws/resource_aws_prometheus_workspace.go

```diff
@@ -0,0 +1,128 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/prometheusservice"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/prometheusservice/waiter"
+)
+
+func resourceAwsPrometheusWorkspace() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAwsPrometheusWorkspaceCreate,
+		ReadContext:   resourceAwsPrometheusWorkspaceRead,
+		UpdateContext: resourceAwsPrometheusWorkspaceUpdate,
+		DeleteContext: resourceAwsPrometheusWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"prometheus_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsPrometheusWorkspaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[INFO] Reading AMP workspace %s", d.Id())
+	conn := meta.(*AWSClient).prometheusserviceconn
+
+	details, err := conn.DescribeWorkspaceWithContext(ctx, &prometheusservice.DescribeWorkspaceInput{
+		WorkspaceId: aws.String(d.Id()),
+	})
+	if tfawserr.ErrCodeEquals(err, prometheusservice.ErrCodeResourceNotFoundException) {
+		log.Printf("[WARN] Prometheus Workspace (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading Prometheus Workspace (%s): %w", d.Id(), err))
+	}
+
+	if details.Workspace != nil {
+		ws := details.Workspace
+		if ws.Arn != nil {
+			d.Set("arn", ws.Arn)
+		}
+		if ws.PrometheusEndpoint != nil {
+			d.Set("prometheus_endpoint", ws.PrometheusEndpoint)
+		}
+		if ws.Alias != nil {
+			d.Set("alias", ws.Alias)
+		}
+	}
```

Comment:
Nit: `(ResourceData).Set()` automatically handles `nil` pointers and we can return an error early to remove the nesting.

```suggestion
	if details == nil || details.Workspace == nil {
		return diag.FromErr(fmt.Errorf("error reading Prometheus Workspace (%s): empty response", d.Id()))
	}

	ws := details.Workspace

	d.Set("alias", ws.Alias)
	d.Set("arn", ws.Arn)
	d.Set("prometheus_endpoint", ws.PrometheusEndpoint)

```